server/services/store/sqlstore/migrate.go

```diff
@@ -282,3 +297,365 @@ func (s *SQLStore) ensureMigrationsAppliedUpToVersion(engine *morph.Morph, drive
 
 	return nil
 }
+
+func (s *SQLStore) GetTemplateHelperFuncs() template.FuncMap {
+	funcs := template.FuncMap{
+		"addColumnIfNeeded":    s.genAddColumnIfNeeded,
+		"dropColumnIfNeeded":   s.genDropColumnIfNeeded,
+		"createIndexIfNeeded":  s.genCreateIndexIfNeeded,
+		"renameTableIfNeeded":  s.genRenameTableIfNeeded,
+		"renameColumnIfNeeded": s.genRenameColumnIfNeeded,
+		"doesTableExist":       s.doesTableExist,
+		"doesColumnExist":      s.doesColumnExist,
+	}
+	return funcs
+}
```

Comment:
I can't do `"addColumnIfNeeded":    s.genAddColumnIfNeeded(schemaName),` since the map value must be a function pointer, not the result of the function call.  I can wrap the callback in a closure.  I also means we can never have scripts that migration more than one schema.

Reply:
Ok, I added the schema name to database store properties.